### ISSUE TITLE
Google Maps Caching and Reject Ambiguous Addresses

### ISSUE THREAD

**Issue by selfuntitled** [SalesforceFoundation/NPSP#2042]
Any address that is validated when "Reject Ambiguous Addresses" is set to true has results cached so, even after "Reject Ambiguous Addresses" is turned back off, the (often incorrect) validation result is persistently stored and returned instead of the correct, validated address.

Steps to reproduce -
Enable address verification using Google API
Set "Reject Ambiguous Addresses" to true
Identify two test contacts:
Create a contact with a valid address and validate that address
Edit a contact with a valid address and then validate the address
Odds are, after validation, both addresses will be updated to invalid or ambiguous addresses.

Set "Reject ambiguous addresses" to false
Edit both contacts, and reset addresses to previously valid addresses.
Save records

Result - both contacts will be re-set to invalid or ambiguous address.

Further test:
Add the valid address from either test contact to a different household and validate address
Result - invalid or ambiguous address stored.

In short, Reject Ambiguous Addresses, seems to permanently break validation on any address it validates, even after it's set back to false.

**Comment by davidhabib** [comment]
**lurch: add

**Comment by LurchTheButler** [comment]
Tracking <a href="https://foundation.my.salesforce.com/a2x80000003RqZHAA0" target="_blank">W-009905</a>

**Comment by davidhabib** [comment]
hub thread here: https://powerofus.force.com/0D58000002Uh6UR

**Comment by mbeaty-sf** [comment]
Salesforce has determined that this issue will not be explored further.
